docs/guides/testing/testing.rst

```diff
@@ -117,144 +117,47 @@ Note -
-
-    paver test_system -s lms --enable-migrations
-
-To see the migration output, use::
-
-    paver test_system -s lms --enable-migrations --verbose --disable_capture
-
-To run a single django test class use this command::
-
-    paver test_system -t lms/djangoapps/courseware/tests/tests.py::ActivateLoginTest
 
 Running a Single Test
 ~~~~~~~~~~~~~~~~~~~~~
 
-When developing tests, it is often helpful to be able to really just run
-one single test without the overhead of PIP installs, UX builds, etc. In
-this case, it is helpful to look at the output of paver, and run just
-the specific command (optionally, stripping away coverage metrics). At
-the time of this writing, the command is the following::
+Various ways to run tests using pytest::
 
-    pytest lms/djangoapps/courseware/tests/test_courses.py
-
-
-To run a single test format the command like this::
-
-    paver test_system -t lms/djangoapps/courseware/tests/tests.py::ActivateLoginTest::test_activate_login
-
-You can use ``--randomize`` to randomize the test case sequence.  In the
-short term, this is likely to reveal bugs in our test setup and teardown;
-please fix (or at least file tickets for) any such issues you encounter.
-
-You can also enable test concurrency with the ``--processes=N`` flag (where ``N``
-is the number of processes to run tests with, and ``-1`` means one process per
-available core). Note, however, that when running concurrently, breakpoints may
-not work correctly.
-
-For example::
-
-    # This will run all tests in the order that they appear in their files, serially
-    paver test_system -s lms --no-randomize --processes=0
-
-    # This will run using only 2 processes for tests
-    paver test_system -s lms --processes=2
-
-To re-run all failing django tests from lms or cms, use the
-``--failed``,\ ``-f`` flag (see note at end of section)::
-
-    paver test_system -s lms --failed
-    paver test_system -s cms --failed
-
-There is also a ``--exitfirst``, ``-x`` option that will stop pytest
-after the first failure.
-
-common/lib tests are tested with the ``test_lib`` task, which also
-accepts the ``--failed`` and ``--exitfirst`` options::
-
-    paver test_lib -l common/lib/xmodule
-    paver test_lib -l common/lib/xmodule --failed
+    pytest path/test_m­odule.py                          # Run all tests in a module.
+    pytest path/test_m­odule.p­y:­:te­st_func               # Run a specific test within a module.
+    pytest path/test_m­odule.p­y:­:Te­stC­las­s               # Run all tests in a class
+    pytest path/test_m­odule.p­y:­:Te­stC­las­s::­tes­t_m­ethod  # Run a specific method of a class.
+    pytest path/testing/                                # Run all tests in a directory.
 
 For example, this command runs a single python unit test file::
 
     pytest common/lib/xmodule/xmodule/tests/test_stringify.py
 
+Various tools like ddt create tests with very complex names, rather than figuring out the name yourself, you can:
```

Comment:
Unfinished example here? Or needs to be joined with the next sentence?

Reply:
There are two examples of how to do this below, but I couldn't figure out the best way to format this

Reply:
nvm, I think I found better formatting